src/screens/Settings/components/ChangePasswordDialog.tsx

```diff
@@ -0,0 +1,314 @@
+    } catch (e: any) {
+      logger.warn('Failed to set new password', {error: e})
+      if (isNetworkError(e)) {
+        setError(
+          _(
+            msg`Unable to contact your service. Please check your internet connection and try again.`,
+          ),
+        )
+      } else {
+        setError(cleanError(e))
+      }
+    } finally {
+      setIsProcessing(false)
+    }
+  }
+
+  const onBlur = () => {
+    const formattedCode = checkAndFormatResetCode(resetCode)
+    if (!formattedCode) {
+      return
+    }
+    setResetCode(formattedCode)
+  }
+
+  return (
+    <Dialog.ScrollableInner
+      label={_(msg`Change password dialog`)}
+      style={web({maxWidth: 450})}>
+      <Dialog.Close />
+      <View style={[a.gap_xl]}>
+        <View style={[a.gap_sm]}>
+          <Text style={[a.font_heavy, a.text_2xl]}>
+            {uiStrings[stage].title}
+          </Text>
+          {error ? (
+            <View style={[a.rounded_sm, a.overflow_hidden]}>
+              <ErrorMessage message={error} />
+            </View>
+          ) : null}
+
+          <Text style={[a.text_md, a.leading_snug]}>
+            {uiStrings[stage].message}
+          </Text>
+        </View>
+
+        {stage === Stages.ChangePassword && (
+          <View style={[a.gap_md]}>
+            <View>
+              <TextField.LabelText>
+                <Trans>Confirmation code</Trans>
+              </TextField.LabelText>
+              <TextField.Root>
+                <TextField.Input
+                  label={_(msg`Confirmation code`)}
+                  placeholder="XXXXX-XXXXX"
+                  value={resetCode}
+                  onChangeText={setResetCode}
+                  onBlur={onBlur}
+                  autoCapitalize="none"
+                  autoCorrect={false}
+                  autoComplete="one-time-code"
+                />
+              </TextField.Root>
+            </View>
+            <View>
+              <TextField.LabelText>
+                <Trans>New password</Trans>
+              </TextField.LabelText>
+              <TextField.Root>
+                <TextField.Input
+                  label={_(msg`New password`)}
+                  placeholder={_(msg`At least 8 characters`)}
+                  value={newPassword}
+                  onChangeText={setNewPassword}
+                  secureTextEntry
+                  autoCapitalize="none"
+                  autoComplete="new-password"
+                />
+              </TextField.Root>
+            </View>
+          </View>
+        )}
+
+        <View style={[a.gap_sm, gtMobile && [a.flex_row_reverse, a.ml_auto]]}>
+          {stage === Stages.RequestCode ? (
+            <>
+              {!gtMobile && (
+                <Button
+                  label={_(msg`Already have a code?`)}
+                  onPress={() => setStage(Stages.ChangePassword)}
+                  size="large"
+                  color="primary"
+                  variant="ghost"
```

Comment:
We can ditch the `ghost` (and all variants on these buttons, actually) and maybe use `primary_subtle`? We're considering `variant` props deprecated moving forward.